build-tools/tasks/unit.js

```diff
@@ -5,17 +5,21 @@ const glob = require('glob');
 const execa = require('execa');
 const { task } = require('../utils/gulp-utils.js');
 
-function runJest(name, configFile, testPattern) {
+function runJest(name, configFile, testPattern, envOverride) {
   return task(`jest:${name}`, async () => {
     // Jest's file matching logic is different than usual globs: https://github.com/facebook/jest/issues/7108
     // We use glob module to provide specific files
     const files = glob.sync(testPattern);
-    await execa('jest', ['-c', configFile, ...files], { stdio: 'inherit' });
+    await execa('jest', ['-c', configFile, ...files], { stdio: 'inherit', env: { ...process.env, ...envOverride } });
   });
 }
 
 // jest has parallellization built-in, no need for concurrency there
 module.exports = series(
-  runJest('build-tools', 'jest.build-tools.config.js', 'build-tools/**/__tests__/*.test.js'),
+  runJest('build-tools', 'jest.build-tools.config.js', 'build-tools/**/__tests__/*.test.js', {
+    // stylelint rules require ESM
+    // https://stylelint.io/migration-guide/to-16#jest-preset-stylelint
+    NODE_OPTIONS: '--experimental-vm-modules',
```

Comment:
Yes, this is experimental, but it is the only way to run ESM in Jest. 

Another option would be to move these tests to Vitest and start gradual Jest removal from this project.